lib/ansible/module_utils/crypto.py

```diff
@@ -341,50 +345,1128 @@ def remove(self, module):
+    '2.23.42.5': ('set-policy', ),
+    '2.23.42.5.0': ('set-policy-root', ),
+    '2.23.42.7': ('certificate extensions', 'set-certExt'),
+    '2.23.42.7.0': ('setCext-hashedRoot', ),
+    '2.23.42.7.1': ('setCext-certType', ),
+    '2.23.42.7.2': ('setCext-merchData', ),
+    '2.23.42.7.3': ('setCext-cCertRequired', ),
+    '2.23.42.7.4': ('setCext-tunneling', ),
+    '2.23.42.7.5': ('setCext-setExt', ),
+    '2.23.42.7.6': ('setCext-setQualf', ),
+    '2.23.42.7.7': ('setCext-PGWYcapabilities', ),
+    '2.23.42.7.8': ('setCext-TokenIdentifier', ),
+    '2.23.42.7.9': ('setCext-Track2Data', ),
+    '2.23.42.7.10': ('setCext-TokenType', ),
+    '2.23.42.7.11': ('setCext-IssuerCapabilities', ),
+    '2.23.42.8': ('set-brand', ),
+    '2.23.42.8.1': ('set-brand-IATA-ATA', ),
+    '2.23.42.8.4': ('set-brand-Visa', ),
+    '2.23.42.8.5': ('set-brand-MasterCard', ),
+    '2.23.42.8.30': ('set-brand-Diners', ),
+    '2.23.42.8.34': ('set-brand-AmericanExpress', ),
+    '2.23.42.8.35': ('set-brand-JCB', ),
+    '2.23.42.8.6011': ('set-brand-Novus', ),
+    '2.23.43': ('wap', ),
+    '2.23.43.1': ('wap-wsg', ),
+    '2.23.43.1.4': ('wap-wsg-idm-ecid', ),
+    '2.23.43.1.4.1': ('wap-wsg-idm-ecid-wtls1', ),
+    '2.23.43.1.4.3': ('wap-wsg-idm-ecid-wtls3', ),
+    '2.23.43.1.4.4': ('wap-wsg-idm-ecid-wtls4', ),
+    '2.23.43.1.4.5': ('wap-wsg-idm-ecid-wtls5', ),
+    '2.23.43.1.4.6': ('wap-wsg-idm-ecid-wtls6', ),
+    '2.23.43.1.4.7': ('wap-wsg-idm-ecid-wtls7', ),
+    '2.23.43.1.4.8': ('wap-wsg-idm-ecid-wtls8', ),
+    '2.23.43.1.4.9': ('wap-wsg-idm-ecid-wtls9', ),
+    '2.23.43.1.4.10': ('wap-wsg-idm-ecid-wtls10', ),
+    '2.23.43.1.4.11': ('wap-wsg-idm-ecid-wtls11', ),
+    '2.23.43.1.4.12': ('wap-wsg-idm-ecid-wtls12', ),
 }
+# #####################################################################################
+# #####################################################################################
 
-_NORMALIZE_NAMES = {
-    'CN': 'commonName',
-    'C': 'countryName',
-    'L': 'localityName',
-    'ST': 'stateOrProvinceName',
-    'street': 'streetAddress',
-    'O': 'organizationName',
-    'OU': 'organizationalUnitName',
-    'SN': 'surname',
-    'GN': 'givenName',
-    'UID': 'userId',
-    'userID': 'userId',
-    'DC': 'domainComponent',
-    'jurisdictionC': 'jurisdictionCountryName',
-    'jurisdictionL': 'jurisdictionLocalityName',
-    'jurisdictionST': 'jurisdictionStateOrProvinceName',
-    'serverAuth': 'TLS Web Server Authentication',
-    'clientAuth': 'TLS Web Client Authentication',
-    'codeSigning': 'Code Signing',
-    'emailProtection': 'E-mail Protection',
-    'timeStamping': 'Time Stamping',
-    'OCSPSigning': 'OCSP Signing',
-}
+_OID_LOOKUP = dict()
+_NORMALIZE_NAMES = dict()
 
 for dotted, names in _OID_MAP.items():
-    for name in names[1:]:
-        _NORMALIZE_NAMES[name] = names[0]
+    for name in names:
+        _NORMALIZE_NAMES[name.lower()] = names[0]
```

Comment:
Maybe something along these lines might be useful?

```suggestion
        assert name.lower() not in _NORMALIZE_NAMES or _NORMALIZE_NAMES[name.lower()] == names[0]
        _NORMALIZE_NAMES[name.lower()] = names[0]
```

Reply:
Ah, those just map the shorter names to the longer ones...

Reply:
Usually yes. If there aren't collisions, such as `uid` and `UID`...

There's also another collision: both `1.3.6.1.7` (`'Mail'`) and `0.9.2342.19200300.100.1.3` (`'rfc822Mailbox', 'mail'`) have the lowercase name `mail`. Besides these two, there are no more collisions.

Reply:
Usually yes. If there aren't collisions, such as `uid` and `UID`...

There's also another collision: both `1.3.6.1.7` (`'Mail'`) and `0.9.2342.19200300.100.1.3` (`'rfc822Mailbox', 'mail'`) have the lowercase name `mail`. Besides these two, there are no more collisions.